src/lib/indexer/__tests__/eventProcessor.test.ts

```diff
@@ -315,4 +315,45 @@ describe('eventProcessor', function () {
       })
     })
   })
+
+  describe('match2-cancel', function () {
```

Comment:
can we add a test for the case where we have a transaction please

Reply:
to be fair have not gone through eventProcessor unit tests, will do thanks for a flag up.

Reply:
This has been updated, when you get a moment would be nice to get your eyes on this